packages/sources/wrapped/src/endpoint/deposits.ts

```diff
@@ -0,0 +1,46 @@
+import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig, InputParameters, AxiosResponse } from '@chainlink/types'
+
+export const supportedEndpoints = ['deposits']
+
+export type Addresses = {
+  address: string
+}
+
+const customError = (data: any) => data.Response === 'Error'
+
+export const inputParameters: InputParameters = {
+  symbol: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const symbol = validator.validated.data.symbol || 'ETH'
+  const url = `/deposits`
+
+  const options = { ...config.api, url }
+  const response = await Requester.request(options, customError)
+  const addresses = response.data[symbol]
+  const keys = Object.keys(response.data).join()
+
+  if (!addresses) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Input, at 'symbol' path, must be one of the following values: ${keys}`,
+      statusCode: 400,
+    })
+  }
+
+  const result = addresses.map((element: string) => {
+    return { address: element }
```

Comment:
What blockchain and network is this planned to be used with?

The default that Proof of Reserves uses is `btc` `mainnet`. If this differs we will need to specify a `coin` and `network`.

Reply:
for now we do not have support with all networks, we are using the `wrapped` with eth by default in conjunction with the` eth_balance` adapter to get the balances